lucene/queries/src/java/org/apache/lucene/queries/function/ValueSourceScorer.java

```diff
@@ -94,4 +104,12 @@ public float getMaxScore(int upTo) throws IOException {
     return Float.POSITIVE_INFINITY;
   }
 
+  /**
+   * Used to externally set a mutable cost for this instance. If set, this cost gets preference over the FunctionValues's cost
+   *
+   * @lucene.experimental
+   */
+  public void setExternallyMutableCost(float cost) {
```

Comment:
Why not simply setMatchCost ?  It's apparent it's "mutable", and it's public and thus "externally" so :-)
I think the javadocs should reference TPI.matchCost.